src/index.ts

```diff
@@ -171,12 +171,44 @@ export class StorageProvider {
     }
 
     // Fetch chain id from provider
-    let network = await this.provider.getNetwork();
+    const network = await this.provider.getNetwork();
+    const hexChainID = '0x' + this.chainId.toString(16)
 
     // throw error if chain id is not equal to the chain id of the app
     if (this.chainId !== network.chainId) {
-      throw new Error('wrong_network, please change the network to the arcana');
+      try {
+        await this.provider.provider.request({
+          method: 'wallet_switchEthereumChain',
+          params: [{ chainId: hexChainID }],
+        });
+      } catch (e) {
+        // This error code indicates that the chain has not been added to the wallet.
+        if (e.code === 4902) {
+          await this.provider.provider.request({
+            method: 'wallet_addEthereumChain',
+            params: [
+              {
+                chainId: hexChainID,
+                chainName: 'Arcana',
+
+                rpcUrls: [chainIdToRPCURL.get(this.chainId)],
+                // blockExplorerUrls: ['https://testnet.bscscan.com'],
```

Comment:
can you remove this commented code?

Or add our explorer endpoint

Reply:
Addressed.